zipper/zipper_test.go

```diff
@@ -154,35 +154,6 @@ func TestMergeValues(t *testing.T) {
 
 			expectedError: errors.Errors{},
 		},
-		{
```

Comment:
I disagree with the reasons behind this commit. This is an OpenSource project and one of the burdens that comes with it - you must take into account what community wants. So I'll consider this as a regression that blocks this PR.

Merge logic *must* take into account different situations for the merging, as it's one of the most annoying problems outside B

Reply:
I do agree that we should think of ways to merge timeseries with different step time, but I disagree that we were doing that correctly before.

The commit I made reverts the situation back to before commit [f5a4fde940](https://github.com/go-graphite/carbonapi/commit/f5a4fde940#diff-0148271788c61e73bbce09e6a68e28c5L254), where we didn't attempt to merge metrics in that case. That commit introduced the new behavior, which has a bug:

In the merging function in current master, we take the metric of lower resolution and upsample its points to the resolution of the other metric. We do this by taking each bucket in the lower-resolution metric and expanding it to buckets of the size of the higher resolution metric. An example is helpful: Say `m1` has secondly resolution and we have 60 seconds of data, and `m2` has minutely resolution and we have 1 minute of data. We can represent these as two arrays of floats `m1 = [ a_1, ..., a_60 ]` and `m2 = [ b_1 ]`. We then expand `m2` to the array `m2' = [ c_1, ..., c_60 ]`, where `c_j = b_1` for all `j`.

For this to be correct, the aggregation function of the metric cannot be 'sum'. If it is, we have now corrupted the user's data, as `m2'` no longer downsamples to `m2`.

I think the right thing to do in this case is to revert to the previous behavior, which didn't try to deal with this case, but did the right thing. We should then open an issue and discuss how to implement merging of metrics with different time steps. As I wrote in the commit message, we have some thoughts on that over here and are not convinced this is as trivial as it might seem at first.

Reply:
We are talking about zipper here, right? And the sole purpose of zipper was to actually heal metrics from corruptions. Upscale metric will not have any gaps, so it will be always healing from upscaled metric and will create artificial data points. Reverting to original behavior is right thing to do here. We can discuss how to deal with such cases later if you feeling that there is such.